spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -253,10 +264,10 @@ public long executePartitionedDmlStatement(Statement statement) {
 	@Override
 	public <T> List<T> queryAll(Class<T> entityClass,
 			SpannerPageableQueryOptions options) {
-		SpannerPersistentEntity<?> persistentEntity = this.mappingContext
-				.getPersistentEntity(entityClass);
+		SpannerPersistentEntity<?> persistentEntity = this.mappingContext.getPersistentEntity(entityClass);
+		String condition = persistentEntity.hasWhere() ? " WHERE " + persistentEntity.getWhere() : "";
```

Comment:
I see this logic is repeated multiple times. We should introduce a method (let's call it `buildWhere`) that checks if a string is empty or null and if not, returns `" WHERE " + string`, otherwise returns `""`